rest/src/main/java/io/edpn/backend/rest/configuration/BeanConfig.java

```diff
@@ -12,21 +13,27 @@
 
 @Configuration
 public class BeanConfig {
+
     @Bean
-    public GetStationService getStationService() {
-        return new GetStationService();
+    public SystemRepository systemRepository(SystemMapper systemMapper) {
+        return new io.edpn.backend.rest.infrastructure.persistence.repository.system.SystemRepository(systemMapper);
+    }
+
+    @Bean
+    public GetSystemUseCase getSystemUseCase(SystemRepository systemRepository) {
+        return new GetSystemService(systemRepository);
     }
 
     @Bean
-    public GetStationUseCase getStationUsecase(GetStationService getStationService) {
-        return getStationService;
+    public GetStationUseCase getStationService() {
+        return new GetStationService();
```

Comment:
this name is a bit fishy... also why are these defined in a config instead of being annotated with `@Service`?

Reply:
It is a "best practice" I picked up from working in an environment where multiple implementations were available for each interface.

The `@Service` annotation works, but it obfuscates what implementations you are actually using behind "Spring Magic"

Reply:
The @Service annotation, in the context of Spring, links the technical infrastructure (the framework) to the domain logic. The annotation is used to indicate that a particular class represents a service component within the application.
In a DDD approach, it is generally recommended to keep the domain logic separate from technical infrastructure concerns. This separation helps to maintain a clear and focused understanding of the business domain without being tightly coupled to specific implementation details.

Reply:
this is a more acceptable explanation. But if we go full on DDD, shouldn't the rest project be modularized further? To create a modular monolith that can be split of easily into microservices if the need ever arises? Right now I see coupling in the domain that is not needed within the bounded contexts. But that discussion goes a bit out of scope for this PR.